src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2261,12 +2261,51 @@ impl Cage {
         }
     }
 
-    //------------------------------------CHMOD HELPER FUNCTION------------------------------------
+    /// ### Description
+    ///
+    /// The `chmod_syscall()` changes a file's mode bits that consist of read, write, and execute file permission bits.
+    /// Changing set-user-ID, set-group-ID, and sticky bits is currently not supported.
+    ///
+    /// ### Arguments
+    ///
+    /// The `chmod_syscall()` accepts two arguments:
+    /// * `path` - pathname of the file whose mode bits we are willing to change (symbolic links are currently 
+    /// not supported). If the pathname is relative, then it is interpreted relative to the current working directory 
+    /// of the calling process.
+    /// * `mode` - the new file mode, which is a bit mask created by bitwise-or'ing zero or more valid mode bits.
+    /// Some of the examples of such bits are `S_IRUSR` (read by owner), `S_IWUSR` (write by owner), 
+    /// `S_ISUID` (set-user-ID), etc.
+    ///
+    /// ### Returns
+    /// 
+    /// Upon successful completion, zero is returned.
+    /// In case of a failure, -1 is returned, and `errno` is set depending on the error, e.g. EACCES, ENOENT, etc.
+    ///
+    /// ### Errors and Panics
+    ///
+    /// Currently, only two errors are supposrted:
+    /// * `EINVAL` - the value of the mode argument is invalid 
+    /// * `ENOENT` - a component of path does not name an existing file
+    /// Other errors, like `EFAULT  , `ENOTDIR`, etc. are not supported.
+    ///
+    /// There are no cases where this syscall panics.
+    ///
+    /// To learn more about the syscall, valid mode bits, and error values, see
+    /// [chmod(2)](https://man7.org/linux/man-pages/man2/chmod.2.html)
 
     pub fn _chmod_helper(inodenum: usize, mode: u32) {
+        //getting a mutable reference to an inode struct that corresponds to the file whose mode bits we want to change
         let mut thisinode = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
+        //log is used to store all the changes made to the filesystem. After the cage is closed, all the collected
+        //changes are serialized and the state of the underlying filsystem is persisted. This allows us to avoid
+        //serializing and persisting filesystem state after every chmod_syscall().
         let mut log = true;
+        //This sanity check of the `mode` argument has already been done in `chmod_syscall()` before this helper funciton is called.
+        //Isn't it redundant to check the same thing here again?
         if mode & (S_IRWXA | (S_FILETYPEFLAGS as u32)) == mode {
```

Comment:
I don't think chmod allows for modifying the file type directly. And even it allows, I don't think the mode assignment statement is correct, since it does not clear the S_FILETYPEFLAGS bits before making a bitwise or to it, which might result in broken S_FILETYPEFLAGS bits.

Reply:
There is no mentioning anywhere that chmod is used to change file type. It is used for changing access mode bits. S_FILETYPEFLAGS is only used as a sanity check to make sure that we are working with the right types of files. This is also mentioned in the comments

Reply:
I actually feel like modifying `if mode & (S_IRWXA | (S_FILETYPEFLAGS as u32)) == mode` to not include S_FILETYPEFLAGS in the first place might be a better choice, but I'll let more professional guys commenting about this
And I guess it might be good to add the test under `ut_lind_fs_chmod_invalid_args` to test for chmod with file type bits passed as argument

Reply:
Looking at the standard now, by design, even if you are not changing the file type, you are not supposed to pass file type bits in the mode argument. It is the task of the underlying syscall to figure out what the file type is and keep it the same after the access mode bits are changed. In other words, mode argument is constructed by or'ing file access mode bits only. I think removing S_FILETYPEFLAGS is a sensible idea.